saml2/saml2-service-provider/src/test/java/org/springframework/security/saml2/provider/service/servlet/filter/Saml2WebSsoAuthenticationRequestFilterTests.java

```diff
@@ -0,0 +1,179 @@
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.security.saml2.Saml2Exception;
+import org.springframework.security.saml2.credentials.Saml2X509Credential;
+import org.springframework.security.saml2.provider.service.registration.RelyingPartyRegistration;
+import org.springframework.security.saml2.provider.service.registration.RelyingPartyRegistrationRepository;
+
+import static java.nio.charset.StandardCharsets.*;
+import static org.mockito.Mockito.*;
+import static org.springframework.security.saml2.credentials.Saml2X509Credential.Saml2X509CredentialType.*;
+
+public class Saml2WebSsoAuthenticationRequestFilterTests {
+
+	private Saml2WebSsoAuthenticationRequestFilter filter;
+	private RelyingPartyRegistrationRepository repository = mock(RelyingPartyRegistrationRepository.class);
+	private MockHttpServletRequest request;
+	private HttpServletResponse response;
+	private MockFilterChain filterChain;
+
+	@Before
+	public void setup() {
+		filter = new Saml2WebSsoAuthenticationRequestFilter(repository);
+		request = new MockHttpServletRequest();
+		response = new MockHttpServletResponse();
+		request.setPathInfo("/saml2/authenticate/registration-id");
+
+		filterChain = new MockFilterChain();
+	}
+
+	@Test
+	public void createSamlRequestRedirectUrlAndReturnUrlWithoutRelayState() throws ServletException, IOException {
+		RelyingPartyRegistration relyingPartyRegistration = RelyingPartyRegistration
+				.withRegistrationId("registration-id")
+				.remoteIdpEntityId("idp-entity-id")
+				.idpWebSsoUrl("sso-url")
+				.assertionConsumerServiceUrlTemplate("template")
+				.credentials(c -> c.add(new Saml2X509Credential(idpPrivateKey(), idpCertificate(), SIGNING, DECRYPTION)))
+				.build();
+
+		when(repository.findByRegistrationId("registration-id"))
+				.thenReturn(relyingPartyRegistration);
+
+		filter.doFilterInternal(request, response, filterChain);
+
+		Assert.assertFalse(response.getHeader("Location").contains("RelayState="));
+	}
+
+	@Test
+	public void createSamlRequestRedirectUrlAndReturnUrlWithRelayState() throws ServletException, IOException {
+		RelyingPartyRegistration relyingPartyRegistration = RelyingPartyRegistration
+				.withRegistrationId("registration-id")
+				.remoteIdpEntityId("idp-entity-id")
+				.idpWebSsoUrl("sso-url")
+				.assertionConsumerServiceUrlTemplate("template")
+				.credentials(c -> c.add(new Saml2X509Credential(idpPrivateKey(), idpCertificate(), SIGNING, DECRYPTION)))
+				.build();
+
+		when(repository.findByRegistrationId("registration-id"))
+				.thenReturn(relyingPartyRegistration);
+
+		request.setParameter("RelayState", "my-relay-state");
+
+		filter.doFilterInternal(request, response, filterChain);
+
+		Assert.assertTrue(response.getHeader("Location").contains("RelayState=my-relay-state"));
+	}
+
+	private static X509Certificate certificate(String cert) {
```

Comment:
We typically break out test support data into a separate class prefixed with `TestNameOfDataWeProvide`

For example you can take a look at: [TestSaml2Credentials](https://github.com/fhanik/spring-security/blob/feature/saml2_login_configurer_tests/config/src/test/java/org/springframework/security/config/annotation/web/configurers/saml2/TestSaml2Credentials.java)

Reply:
Do you think it's ok to reuse data from TestSaml2X509Credentials by making them public (or moving the class) ?

Reply:
For now, if you include a package private class, we can consolidate test support later.

Reply:
Thanks for the review.

I've moved test data to another class.